multicore_runtime/cloud_logging_test.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+import logging
+import os
+import unittest
+
+from cloud_logging import CloudLoggingHandler
+
+class CloudLoggingTestCase(unittest.TestCase):
+
+  EXPECTED_TRACE_ID = None
+  EXPECTED_OVERRIDDEN_TRACE_ID = '1234123412341234'
+  EXPECTED_MESSAGE = 'test message'
+
+  def setUp(self):
+    self.handler = CloudLoggingHandler()
+    self.record = logging.makeLogRecord({'msg': self.EXPECTED_MESSAGE,
+                                         'levelname': 'INFO'})
+    self.record_with_extra = logging.makeLogRecord(
+        {'msg': self.EXPECTED_MESSAGE,
+         'levelname': 'INFO',
+         'trace_id': self.EXPECTED_OVERRIDDEN_TRACE_ID,})
+
+  def test_file_name_is_correct(self):
+    self.assertTrue(self.handler.baseFilename.startswith('/var/log/app_engine/app.'))
+    self.assertTrue(self.handler.baseFilename.endswith('.json'))
+
+  def test_format(self):
+    msg = self.handler.format(self.record)
+    payload = json.loads(msg)
+    if self.EXPECTED_TRACE_ID:
```

Comment:
If I use the two-class system as here, then yes. This method is run twice, once in each class with a different env setup and a different expected result. I think this is idiomatic for unittest style.